File: code_examples/3-Structured_Problem_Solving/3-Assingment_Statements/CPP/main.cpp
// Copyright (c) 2020 Mr. Coxall All rights reserved.
//
// Created by: Mr. Coxall
// Created on: Sep 2020
// This program shows assingment statements

#include <iostream>

int main() {
    // variable definition
    int numberOfStudents = 2;
    float width = 32.5F;
    float length = 10.0F;
    float areaOfRectangle = 0;
    std::string someWords1 = "Hello";
    std::string someWords2 = "Wordl!";
    std::string helloWorld = "";

    // using assignment statements
    numberOfStudents = numberOfStudents + 5;
    areaOfRectangle = length * width;
    helloWorld = someWords1 + ", " + someWords2;

    // output
    std::cout << "The number of students is: " << numberOfStudents << std::endl;
    std::cout << "The area of a rectangle is: " << areaOfRectangle
              << " cm²" << std::endl;
    std::cout << helloWorld << std::endl;

    std::cout << "\nDone." << std::endl;
}

File: code_examples/3-Structured_Problem_Solving/3-Assingment_Statements/CSharp/main.cs
/* Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program shows assingment statements
*/

using System;

/*
 * The Program class
 * Contains all methods for performing basic variable usage
*/
class Program {
    public static void Main (string[] args) {
        // variable definition
        int numberOfStudents = 2;
        float width = 32.5F;
        float length = 10.0F;
        float areaOfRectangle = 0F;
        string someWords1 = "Hello";
        string someWords2 = "World!";
        string helloWorld = null;

        // using assignment statements
        numberOfStudents = numberOfStudents + 5;
        areaOfRectangle = length * width;
        helloWorld = someWords1 + ", " + someWords2; 

        // output
        Console.WriteLine ("The number of students is: " + numberOfStudents);
        Console.WriteLine ("The area of a rectangle is: " + areaOfRectangle + " cm²");
        Console.WriteLine (helloWorld);

        Console.WriteLine ("\nDone.");
    }
}
